crates/turbopack-core/src/resolve/plugin.rs

```diff
@@ -0,0 +1,59 @@
+use anyhow::Result;
+use turbo_tasks::primitives::BoolVc;
+use turbo_tasks_fs::{glob::GlobVc, FileSystemPathVc};
+
+use crate::resolve::{parse::RequestVc, ResolveResultVc};
+
+#[turbo_tasks::value]
+pub struct ResolvePluginCondition {
+    root: FileSystemPathVc,
+    glob: GlobVc,
+}
+
+#[turbo_tasks::value_impl]
+impl ResolvePluginConditionVc {
+    #[turbo_tasks::function]
+    pub fn new(root: FileSystemPathVc, glob: GlobVc) -> Self {
+        ResolvePluginCondition { root, glob }.cell()
+    }
+
+    #[turbo_tasks::function]
+    pub async fn matches(self, fs_path: FileSystemPathVc) -> Result<BoolVc> {
+        let this = self.await?;
+        let root = this.root.await?;
+        let glob = this.glob.await?;
+
+        let path = fs_path.await?;
+
+        if let Some(path) = root.get_path_to(&path) {
+            if glob.execute(path) {
+                return Ok(BoolVc::cell(true));
+            }
+        }
+
+        Ok(BoolVc::cell(false))
+    }
+}
+
+#[turbo_tasks::value(transparent)]
+pub struct ResolveResultOption(Option<ResolveResultVc>);
+
+#[turbo_tasks::value_impl]
+impl ResolveResultOptionVc {
+    #[turbo_tasks::function]
+    pub fn some(result: ResolveResultVc) -> Self {
+        ResolveResultOption(Some(result)).cell()
+    }
+
+    #[turbo_tasks::function]
+    pub fn none() -> Self {
+        ResolveResultOption(None).cell()
+    }
+}
+
+#[turbo_tasks::value_trait]
+pub trait ResolvePlugin {
```

Comment:
Should this be a `ResolvePlugin` or a straight up `Resolver`? I'm asking because in https://github.com/vercel/turbo/blob/89ead9b9d4a78b27c3eb100b4de580ed7ddd82c0/crates/turbopack-core/src/resolve/mod.rs#L576 we already have three clear passes:
1. `ImportMap(Resolver)`
2. `Request(Resolver)`
3. `FallbackImportMap(Resolver)`

And so it seems like this would essentially add intermediate resolvers between 2. and 3.

Reply:
Not sure if I understand the question, what's a `Resolver`? Would that be a new type?

The `ResolvePlugin` right now is kinda part of pass 2, we already have the `resolved_map` in there.
And it would later on also become part of pass 1

Was thinking the import map could be implemented as a `ResolvePlugin` (not that we would necessarily want that)

Reply:
`Resolver` is just another name for `ResolvePlugin`. I'm saying the current resolving logic can be separated into 3 `ResolvePlugins`.